tools/code_format/check_format.py

```diff
@@ -1080,6 +1082,18 @@ def whitelisted_for_memcpy(self, file_path):
         return file_path in MEMCPY_WHITELIST
 
 
+def normalize_path(path):
+    """Convert path to form ./path/to/dir/ for directories and ./path/to/file otherwise"""
+    isdir = os.path.isdir(args.target_path)
+    if not path.startswith("./"):
```

Comment:
Would it make sense to prepend "./" before checking if it is a directory? Just in case.

Reply:
im wondering if `pathlib` could be helpful here - it does quite a bit of path normalization - altho it seems like elsewhere is interpreting the appended `/` so maybe that just complicates things

Reply:
> Would it make sense to prepend "./" before checking if it is a directory? Just in case.

@yanavlasov - Good point, calling this with an absolute path would lead to unexpected behaviour

> im wondering if `pathlib` could be helpful here - it does quite a bit of path normalization - altho it seems like elsewhere is interpreting the appended `/` so maybe that just complicates things

@phlax - I think in the future something like `pathlib` could be quite helpful, but I think the introduction of path normalization via `pathlib` or some other library function should come with a larger refactor in this file. In particular, I think a diff changing to normalize via a library would be a good opportunity to refactor how paths are handled in general across the formatter. 

As you mentioned, the trailing slash becomes a bit awkward to deal with as we would be modifying the path after it is normalized - hopefully this could be solved with such an additional refactor